include/range/v3/range/concepts.hpp

```diff
@@ -188,88 +178,28 @@ namespace ranges
-                                      iter_reference_t<iterator_t<T const>>);
-            }
-            static constexpr auto test(void const *) -> bool
-            {
-                return true;
-            }
-            template<typename T>
-            constexpr enable_view_helper_(T const * p)
-              : result_(enable_view_helper_::test(p))
-            {}
-        };
-        constexpr bool enable_view_impl_(...)
-        {
-            return false;
-        }
-        constexpr bool enable_view_impl_(view_base const *)
-        {
-            return true;
-        }
-        constexpr bool enable_view_impl_(enable_view_helper_ ev)
-        {
-            return ev.result_;
-        }
-        template<typename T>
-        constexpr bool enable_view_impl_(std::initializer_list<T> const *)
-        {
-            return false;
-        }
-        template<typename Key, typename Compare, typename Alloc>
-        constexpr bool enable_view_impl_(std::set<Key, Compare, Alloc> const *)
-        {
-            return false;
-        }
-        template<typename Key, typename Compare, typename Alloc>
-        constexpr bool enable_view_impl_(std::multiset<Key, Compare, Alloc> const *)
-        {
-            return false;
-        }
-        template<typename Key, typename Hash, typename Pred, typename Alloc>
-        constexpr bool enable_view_impl_(
-            std::unordered_set<Key, Hash, Pred, Alloc> const *)
-        {
-            return false;
-        }
-        template<typename Key, typename Hash, typename Pred, typename Alloc>
-        constexpr bool enable_view_impl_(
-            std::unordered_multiset<Key, Hash, Pred, Alloc> const *)
-        {
-            return false;
-        }
-        // BUGBUG TODO
-        // template<typename BidiIter, typename Alloc>
-        // constexpr bool enable_view_impl_(std::match_results<BidiIter, Alloc> const *)
-        // {
-        //     return false;
-        // }
         template<typename T>
-        constexpr T const * nullptr_(int)
-        {
-            return nullptr;
-        }
-        template<typename T>
-        constexpr int nullptr_(long)
-        {
-            return 0;
-        }
+        struct enable_view_
+          : std::is_base_of<view_base, T>
+        {};
     } // namespace detail
     /// \endcond
 
     // Specialize this if the default is wrong.
     template<typename T>
     RANGES_INLINE_VAR constexpr bool enable_view =
-        detail::enable_view_impl_(detail::nullptr_<T>(0));
+        detail::enable_view_<T>::value;
+
+#if defined(__cpp_lib_string_view) && __cpp_lib_string_view > 0
+    template<typename Char, typename Traits>
+    RANGES_INLINE_VAR constexpr bool enable_view<std::basic_string_view<Char, Traits>> =
+        true;
+#endif
+
+#if defined(__cpp_lib_span) && __cpp_lib_span > 0
+    template<typename T, std::size_t N>
+    RANGES_INLINE_VAR constexpr bool enable_view<std::span<T, N>> = true;
```

Comment:
```suggestion
    RANGES_INLINE_VAR constexpr bool enable_view<std::span<T, N>> = N == 0 || N == std::dynamic_extent;
```

Reply:
I really should have worded this as `N + 1 < 2` and told the unsigned type proponents that they asked for it.

(FWIW: the value of `enable_view` doesn't really matter for the specializations of `span` that don't satisfy `semiregular`, I complicated the wording to respect the letter of the wording that says users may specialize `enable_view` to `true` for "types that model `view`".)

Reply:
> I really should have worded this as `N + 1 < 2`

I'm going with that formulation because it doesn't require a declaration of `std::dynamic_extent`.